src/tokenize.c

```diff
@@ -396,6 +396,8 @@ static Token read_token(struct State *st)
                 if (is_keyword(t.data.name))
                     t.type = TOKEN_KEYWORD;
                 else if ('0'<=t.data.name[0] && t.data.name[0]<='9') {
+                    if (is_vaild_float(t.data.name))
+                        t.type = TOKEN_FLOAT;
```

Comment:
To detect floats, you could do something similar to what I did with longs, i.e. look for `f` suffix.

Or maybe just get rid of float literals, at least for this PR: `12.34f` would do the same thing as `12.34 as float` anyway, so it's not really necessary.